tests/TestCase/ORM/EntityTest.php

```diff
@@ -1573,4 +1573,68 @@ public function testIsDirtyFromClone()
         $this->assertTrue($cloned->dirty('a'));
         $this->assertTrue($cloned->dirty('b'));
     }
+
+    /**
+     * Test the isEmpty() check
+     *
+     * @return void
+     */
+    public function testIsEmpty()
+    {
+        $entity = new Entity([
+            'array' => ['foo' => 'bar'],
+            'emptyArray' => [],
+            'object' => new \stdClass(),
+            'string' => 'string',
+            'emptyString' => '',
+            'intZero' => 0,
+            'intNotZero' => 1,
+            'floatZero' => 0.0,
+            'floatNonZero' => 1.5,
+            'null' => null
+        ]);
+
+        $this->assertFalse($entity->isEmpty('array'));
```

Comment:
Maybe group both - the list above and the asset list below - by their return value (true/false)